ios/FluentUI/Navigation/SideBar.swift

```diff
@@ -0,0 +1,284 @@
+		addSubview(borderLine)
+
+		addSubview(topStackView)
+		addSubview(bottomStackView)
+
+		if #available(iOS 13, *) {
+			addInteraction(UILargeContentViewerInteraction())
+		}
+
+		accessibilityTraits = .tabBar
+
+		NSLayoutConstraint.activate([widthAnchor.constraint(equalToConstant: Constants.viewWidth),
+									 borderLine.leadingAnchor.constraint(equalTo: trailingAnchor),
+									 borderLine.bottomAnchor.constraint(equalTo: bottomAnchor),
+									 borderLine.topAnchor.constraint(equalTo: topAnchor)])
+	}
+
+	private var layoutConstraints: [NSLayoutConstraint] = []
+
+	private func setupLayout() {
+		if layoutConstraints.count > 0 {
+			NSLayoutConstraint.deactivate(layoutConstraints)
+			layoutConstraints.removeAll()
+		}
+
+		if let avatarView = self.avatarView {
+			layoutConstraints.append(contentsOf: [
+				avatarView.topAnchor.constraint(equalTo: safeAreaLayoutGuide.topAnchor, constant: 18.0),
+				avatarView.centerXAnchor.constraint(equalTo: centerXAnchor),
+				avatarView.widthAnchor.constraint(equalToConstant: Constants.avatarViewSize),
+				avatarView.heightAnchor.constraint(equalToConstant: Constants.avatarViewSize),
+				topStackView.topAnchor.constraint(equalTo:avatarView.bottomAnchor , constant: 34.0)
+			])
+		} else {
+			layoutConstraints.append(topStackView.topAnchor.constraint(equalTo: safeAreaLayoutGuide.topAnchor, constant: 30.0))
+		}
+
+		layoutConstraints.append(contentsOf: [
+			topStackView.widthAnchor.constraint(equalTo: widthAnchor),
+			topStackView.heightAnchor.constraint(equalToConstant: stackViewHeight(topStackView)),
+			bottomStackView.widthAnchor.constraint(equalTo: widthAnchor),
+			bottomStackView.heightAnchor.constraint(equalToConstant: stackViewHeight(bottomStackView)),
+			bottomStackView.bottomAnchor.constraint(equalTo: safeAreaLayoutGuide.bottomAnchor, constant: -10.0)
+		])
+
+		NSLayoutConstraint.activate(layoutConstraints)
+	}
+
+	private func stackViewHeight(_ stackView: UIStackView) -> CGFloat {
+		let itemCount: CGFloat = CGFloat(stackView.arrangedSubviews.count)
+		var height: CGFloat = 24.0 * itemCount
+
+		if (itemCount > 0) {
+			height += stackView.spacing * (itemCount - 1)
+		}
+
+		return height
+	}
+
+	private func items(in section: Section) -> [TabBarItem] {
+		switch section {
+		case .top:
+			return topItems
+		case .bottom:
+			return bottomItems
+		}
+	}
+
+	private func stackview(in section: Section) -> UIStackView {
+		switch section {
+		case .top:
+			return topStackView
+		case .bottom:
+			return bottomStackView
+		}
+	}
+
+	private func itemView(with item: TabBarItem, in section: Section) -> TabBarItemView? {
+		if let index = items(in: section).firstIndex(of: item) {
+			if let tabBarItemView = stackview(in: section).arrangedSubviews[index] as? TabBarItemView {
+				return tabBarItemView
+			}
+		}
+
+		return nil
+	}
+
+	@objc private func handleAvatarViewTapped(_ recognizer: UITapGestureRecognizer) {
+		delegate?.sideBar?(self, didActivate: avatarView!)
+	}
```

Comment:
Per other comment, can we just let the client put a view in here themselve so they are free to handle it? On the other hand if this control is opinionated enough that the view needs to be an avatar view, then maybe it should be opinionated enough to handle that click ourselves? But we probably shouldn't do one and not the other.

Reply:
As mentioned in my other responses, we want to force this view to be an avatar view that the client can provide. If no avatar view is provided, we don't display anything above the tab bar items.
As for touch events, we are making it easier on clients to intercept touch events by adding our own gesture recognizer. I personally like this approach because it simplifies the usage of this class. Otherwise, each client will need to add duplicated logic to handle touch events on the view. This also allows us to define the touch area for the avatar view. Right now, the touch area is just the view itself but, we could change that if we wanted to.